src/MergeContext.js

```diff
@@ -499,58 +559,115 @@ class MergeContext {
            else
                throw e;
         }
-        // https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref
-        // state is one of 'failure', 'error', 'pending' or 'success'.
-        // We treat both 'failure' and 'error' as an 'error'.
-        let combinedStatus = await GH.getStatuses(ref);
+
         if (requiredContexts === undefined || requiredContexts.length === 0) {
             this._log("no required contexts found");
-            // rely on all available checks then
-            return combinedStatus.state;
-        }
-        // If checksNumber was passed, we use required status context string matching
-        // for required checks counting. Gotten tag checks are compared against those
-        // configured for protected base branch. For simplicity, we use 'starts with'
-        // matching rule.
-        // For example, if we configured three required checks(Config.stagingChecks=3) but
-        // the branch has a single required check named 'Jenkins(build test)',
-        // the bot will wait for three checks with 'Jenkins(build test).*' names.
-        const needMatching = checksNumber !== undefined;
-        const requiredChecksNumber = needMatching ? checksNumber : requiredContexts.length;
-
-        // An array of [{context, state}] elements
-        let requiredChecks = [];
+            return null;
```

Comment:
The caller can check with '!result' instead of '!result.length'. Not a great benefit, but a bit simpler. As I see, callers anyway do this check and return almost immediately.

Reply:
You may be missing my point: There should be no checks in the callers, not for null array, not for an empty array. If we return an empty array, the callers code should work correctly without any special logic dedicated to this case (because it is not so special). If we return `null`, we create a special case that must be treated specially in the callers.